rfcs/20201014-VTune-ITT-tagging/README.md

```diff
@@ -0,0 +1,142 @@
+DNNL_ITT_TASKS=False   
+Moreover, user can give a custom VTune path if they don't have it under /opt/intel/...   
+CMAKE_VTUNE_HOME=<CUSTOM PATH>  
+
+**Runtime**  
+
+Users can use a environment variable to control ITT level at runtime.  
+
+| DNN_ITT   | __itt_task_level  | Note |  
+|---------------|---------------------|---------------------|  
+| 0   | __itt_task_level_none |   no ITT event will be triggered  | 
+| 1 | __itt_task_level_low  |    ITT events are only triggered in master thread     | 
+| 2 | __itt_task_level_high |    ITT events are triggered in all OMP/TBB threads      | 
+
+
+### Testing aspects
+---
+**Performance Number**  
+
+
+We average 20 runs for each number.
+For the last two tests, we made the tests without/with VTune running.
+Due to profiling size limitation, VTune profiling ended very early, so we regarded the number is not applicable.
+
+
+| Primitive     | Benchdnn input file                            | Baseline (no patch) | Patch + no VTune + DNN_ITT=3| Patch + Vtune + DNN_ITT=3 |
+|---------------|------------------------------------------------|---------------------|------------------|---------------|
+| Convolution   | tests/benchdnn/inputs/conv/set_perf_cpu_all_mb | 2858.6515 (stddev : 88.3)    | 2785.1985 ( stddev: 83.13)      | NA (excess 1G data size very quickly)   |
+| Inner Product | tests/benchdnn/inputs/ip/perf_ip_inference_sb  | 349.4827 (stddev : 41.64)         | 349.16535 (stddev : 49.15)   | NA (excess 1G data size very quickly) |
+
+
+For performance numbers with VTune running, we used much smaller tests to get VTune profiling results from start to end.
+
+| Primitive     | Benchdnn input file                            | Baseline (no patch) | Patch + no VTune + DNN_ITT=0| Patch + no VTune + DNN_ITT=3| Patch + Vtune + DNN_ITT=3 |  
+|---------------|------------------------------------------------|---------------------|------------------|------------------|---------------|
+| convolution | tests/benchdnn/inputs/conv/shapes_alexnet        | 79.93 ms | 80.76 ms| 79.34 ms | 88.25 ms | 
+| ip | tests/benchdnn/inputs/ip/shapes_alexnet        | 4.45 ms | 4.598 ms |4.57 ms |  6.002 ms |
+| bnorm | tests/benchdnn/inputs/bnrom/shapes_resnet_50       | 51.22 ms | 51.25 ms|50.82 ms |  68.19 ms |
+| matmul | tests/benchdnn/inputs/matmul/shapes_3d       | 74.7 ms | 81 ms |77.1 ms |  81.17 ms |
+| softmax | tests/benchdnn/inputs/softmax/shapes_2d       | 125.29 ms | 127.9 ms|126.48 ms |  145.96 ms |
+  
+We indeed saw overhead when we have VTune running, and the slow down is around 75% - 90%.  
+However, the performance with/without this feature is almost the same when we don't have VTune running.  
+**In short, we don't see overhead to enable the feature by default without VTune running.**  
+
+
+## Open Questions  
+
+1. Should we rename environent variable DNNL_ITT?
+2. Should we set DNNL_ITT as __itt_task_level_high by default?
```

Comment:
I think there should be the following default options: `DNNL_ENABLE_VTUNE_TAGS OFF`, `DNNL_VTUNE_TAGS_LEVEL LOW`

Reply:
For Tensorflow, they don't use CMakefile, so we need recompile TF with a compilation option.

For PyTorch, I am not sure if they will use Option.CMakefile file.
I need to double check.

I don't see performance degradation if we enable the feature by default. Should we still turn it off by default?

Reply:
Feedback from VTune team : 
"It will be very great to have this ITT instrumentation by default, so that we won’t need to re-compile oneDNN each time we need to do perf analysis which is very inconvenient (same for JIT API, BTW). As your measurements show having it doesn’t lead to any perf regressions if running without VTune, so nothing prevents us from doing it I think."

Reply:
Makes sense, thanks for following up with feedback from VTune team, then we can leave it `ON` by default.